mypy/typeops.py

```diff
@@ -130,6 +141,12 @@ def map_type_from_supertype(typ: Type,
     return expand_type_by_instance(typ, inst_type)
 
 
+def instance_or_var(typ: ProperType) -> bool:
+    # TODO: use more principled check for non-trivial self-types.
```

Comment:
Maybe we should handle `typ` being a TypeType in here as well. That would let us clean up the if statement down below a little more.